python/ray/ml/examples/huggingface/huggingface_text_classification.ipynb

```diff
@@ -0,0 +1,1235 @@
+    "model_name = model_checkpoint.split(\"/\")[-1]\n",
+    "validation_key = \"validation_mismatched\" if task == \"mnli-mm\" else \"validation_matched\" if task == \"mnli\" else \"validation\"\n",
+    "name = f\"{model_name}-finetuned-{task}\"\n",
+    "\n",
+    "def trainer_init_per_worker(train_dataset, eval_dataset = None, **config):\n",
+    "    print(f\"Is CUDA available: {torch.cuda.is_available()}\")\n",
+    "    metric = load_metric_fn()\n",
+    "    tokenizer = load_tokenizer_fn()\n",
+    "    model = AutoModelForSequenceClassification.from_pretrained(model_checkpoint, num_labels=num_labels)\n",
+    "    args = TrainingArguments(\n",
+    "        name,\n",
+    "        evaluation_strategy=\"epoch\",\n",
+    "        save_strategy=\"epoch\",\n",
+    "        learning_rate=2e-5,\n",
+    "        per_device_train_batch_size=batch_size,\n",
+    "        per_device_eval_batch_size=batch_size,\n",
+    "        num_train_epochs=5,\n",
+    "        weight_decay=0.01,\n",
+    "        push_to_hub=False,\n",
+    "        disable_tqdm=True,\n",
+    "        no_cuda=not torch.cuda.is_available(),\n",
+    "    )\n",
+    "\n",
+    "    def compute_metrics(eval_pred):\n",
+    "        predictions, labels = eval_pred\n",
+    "        if task != \"stsb\":\n",
+    "            predictions = np.argmax(predictions, axis=1)\n",
+    "        else:\n",
+    "            predictions = predictions[:, 0]\n",
+    "        return metric.compute(predictions=predictions, references=labels)\n",
+    "\n",
+    "    trainer = Trainer(\n",
+    "        model,\n",
+    "        args,\n",
+    "        train_dataset=train_dataset,\n",
+    "        eval_dataset=eval_dataset,\n",
+    "        tokenizer=tokenizer,\n",
+    "        compute_metrics=compute_metrics\n",
+    "    )\n",
+    "\n",
+    "    print(\"Starting training\")\n",
+    "    return trainer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "CdzABDVcIrJg"
+   },
+   "source": [
+    "With our `trainer_init_per_worker` complete, we can now instantiate the `HuggingFaceTrainer`. Aside from the function, we set the `scaling_config`, controlling the amount of workers and resources used, and the `datasets` we will use for training and evaluation.\n",
+    "\n",
+    "We will use 2 workers, each with a single GPU assigned, and we specify the `MlflowLoggerCallback` inside the `run_config`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ray.ml.train.integrations.huggingface import HuggingFaceTrainer\n",
+    "from ray.ml import RunConfig\n",
+    "from ray.tune.integration.mlflow import MLflowLoggerCallback\n",
+    "\n",
+    "trainer = HuggingFaceTrainer(\n",
+    "    trainer_init_per_worker=trainer_init_per_worker,\n",
+    "    scaling_config={\"num_workers\": 2, \"use_gpu\": True},\n",
```

Comment:
Make the configurability of num_workers more obvious, and explain how this maps to the number of GPUs in the cluster.